### PR TITLE
Optimized BorderedTable, refactored CityButton, new air table for units + misc

### PR DESCRIPTION
**Pull request by vegeta1k95**
1) Rewritten `BorderedTable` class: removed inner Images, override `draw` instead.
Less objects, less bugs, more speed.

2) Chore: Rearranged `CityButton` class and file, separated each plate component in own class: `InfluenceTable`, `DefenceTable`, `CityTable`, etc. No visual changes.

3) New look for air table for units:
imageimageimage

4) Fixes:
- removed bugged selection highlight for Policy buttons in Policy Screen
- fixed wrong alignment for circles of city units
image
- removed redundant "Fortified" action indicator (fortified status is visible via flag shape already)
image to image

### CODE REVIEW

File: core/src/com/unciv/ui/pickerscreens/PolicyPickerScreen.kt [hunk 1]
```diff
@@ -301,9 +293,8 @@ class PolicyPickerScreen(val worldScreen: WorldScreen, civInfo: Civilization = w
 
         // Main table
         val colorBg = if (branch.isAdopted()) PolicyColors.branchAdopted else PolicyColors.branchNotAdopted
-        val branchGroup = BorderedTable(
-            path="PolicyScree/PolicyBranchBackground",
-            innerColor = colorBg)
+        val branchGroup = BorderedTable(path="PolicyScree/PolicyBranchBackground")
```
Comment (OptimizedForDensity) [inline core/src/com/unciv/ui/pickerscreens/PolicyPickerScreen.kt:296]: >"PolicyScree/[...]
Typo but not introduced in this PR.

File: core/src/com/unciv/ui/tilegroups/layers/TileLayerUnitFlag.kt [hunk 2]
```diff
@@ -71,32 +80,28 @@ class TileLayerUnitFlag(tileGroup: TileGroup, size: Float) : TileLayer(tileGroup
     }
 
     private fun getAirUnitTable(unit: MapUnit): Table {
-        val holder = Table()
 
-        val iconColor = unit.civInfo.nation.getInnerColor()
-        val bgColor = unit.civInfo.nation.getOuterColor()
+        val iconColor = unit.civInfo.nation.getOuterColor()  //Color.WHITE.cpy().apply { a = 0.9f }
+        val bgColor = unit.civInfo.nation.getInnerColor() //Color.BLACK.cpy().apply { a = 0.9f }
```
Comment (OptimizedForDensity) [inline core/src/com/unciv/ui/tilegroups/layers/TileLayerUnitFlag.kt:85]: Not something that should be put in this PR but it seems to me that there should be a standard class for icons like this on the world screen. Something like `WorldScreenCivilizationIcon(civ: Civilization, shape: [some shape enum like square or circle or diamond or rounded horizontal rectangle {pill shape?} depending on need])` that looks for the `civ`'s main and secondary colors and can be applied to all sorts of image groups.

File: core/src/com/unciv/ui/tilegroups/CityButton.kt [hunk 3]
```diff
@@ -31,25 +30,334 @@ import com.unciv.ui.utils.extensions.colorFromRGB
...
+
+        if (city.isBeingRazed) {
+            val fireImage = ImageGetter.getImage("OtherIcons/Fire")
+            add(fireImage).size(18f).padLeft(2f)
+        }
+
+        if (city.civInfo == viewingCiv && city.isWeLoveTheKingDayActive()) {
+            val wltkdImage = ImageGetter.getImage("OtherIcons/WLTKD")
+            add(wltkdImage).size(18f).padLeft(2f)
+        }
```
Comment (OptimizedForDensity) [inline core/src/com/unciv/ui/tilegroups/CityButton.kt:186]: Lots of `size(18f).padLeft(2f)` that seems like it should be moved to a constant somewhere.

File: core/src/com/unciv/ui/tilegroups/CityButton.kt [hunk 4]
```diff
@@ -31,25 +30,334 @@ import com.unciv.ui.utils.extensions.colorFromRGB
...
+
+        val normalizedInfluence = max(-60f, min(influence, 60f)) / 30f
+
+        val color = when (relationshipLevel) {
+            RelationshipLevel.Unforgivable -> Color.RED
+            RelationshipLevel.Enemy -> Color.ORANGE
+            RelationshipLevel.Afraid -> Color.YELLOW
+            RelationshipLevel.Neutral, RelationshipLevel.Friend -> Color.LIME
+            RelationshipLevel.Ally -> Color.SKY
+            else -> Color.DARK_GRAY
```
Comment (OptimizedForDensity) [inline core/src/com/unciv/ui/tilegroups/CityButton.kt:57]: Looks like `DiplomacyManager.RelationshipLevel` already has these colors coded so they might be worth reusing.

File: core/src/com/unciv/ui/tilegroups/CityButton.kt [hunk 5]
```diff
@@ -31,25 +30,334 @@ import com.unciv.ui.utils.extensions.colorFromRGB
...
+
+        val textColor = city.civInfo.nation.getInnerColor()
+
+        val cityConstructions = city.cityConstructions
+        val cityCurrentConstruction = cityConstructions.getCurrentConstruction()
+
+        val progressTable = Table()
+
+        var percentage = 0f
+        var turns = "∞"
```
Comment (OptimizedForDensity) [inline core/src/com/unciv/ui/tilegroups/CityButton.kt:325]: Reminds me that this character should probably be a constant but that's a random thought and not a critique.

File: core/src/com/unciv/ui/tilegroups/layers/TileLayerUnitFlag.kt (hunk 2 above)
Comment (vegeta1k95) [reply core/src/com/unciv/ui/tilegroups/layers/TileLayerUnitFlag.kt:85]: World scree Units icons are visualized already by `UnitGroup` class, which selects shape and color based on type/status/civ.

File: core/src/com/unciv/ui/tilegroups/CityButton.kt (hunk 4 above)
Comment (vegeta1k95) [reply core/src/com/unciv/ui/tilegroups/CityButton.kt:57]: Those colors are different, they are used for diplomacy/leader screen.
E.g. `Ally` color for diplomacy is green, while `Ally` indicator for influence bar should be blue.

File: core/src/com/unciv/ui/pickerscreens/PolicyPickerScreen.kt (hunk 1 above)
Comment (vegeta1k95) [reply core/src/com/unciv/ui/pickerscreens/PolicyPickerScreen.kt]: fixed

**Comment by OptimizedForDensity** [comment]
Do you by chance have a before/after screenshot with a bunch of units in the picture? I'm curious to see how the changes look when a game is played out and the screen is covered with units. Everything looks good otherwise.

**Comment by vegeta1k95** [comment]
Before:
image
After:
image

**Pull request merged**